PLUGINS.md

```diff
@@ -0,0 +1,180 @@
+# OpenShift Console Plugins
+
+Openshift Console Plugins extend the functionality of the console. Plugins are enabled by
+adding them as `dependencies` of the [console-app](frontend/packages/console-app) package.
+
+See the [demo plugin](frontend/packages/console-demo-plugin) for reference.
+
+## Directory and File Names
+
+Plugins packages should be in [frontend/packages](frontend/packages) directory. A plugin should have
+a `package.json`, `OWNERS`, and a plugin entry file, the plugin entry file is usually
+called `src/plugin.tsx`.
+
+Optionally unit tests will be in `__tests__` directories close to the source code, and 
+integration tests should be in an `integration-tests` directory.
+
+``` bash
+    .
+    ├── ...
+    ├── frontend
+    │   ├── packages
+    |   |   ├── ...
+    │   |   ├── my-plugin
+    |   |   |   ├── integration-tests
+    |   |   |   |   └── ...
+    |   |   |   ├── src
+    |   |   |   |   ├── plugin.tsx
+    |   |   |   |   ├── __tests__
+    |   |   |   |   |   ├── some-test.spec.ts
+    |   |   |   |   |   └── ...
+    |   |   |   |   └── ...
+    |   |   |   ├── OWNERS
+    |   |   |   ├── package.json
+    |   |   |   └── ...
+    |   |   └── ...
+    │   └── ...
+    └── ...
+```
+
+## Plugin Directory
+
+Plugin directory name should match the plugin name (console plugins are an exception), as a convention plugin 
+directory names should end with the `-plugin` suffix (eg. `my-plugin`).
+
+- Use lowercase dash-separated names for all files (to avoid git issues with case-insensitive file systems)
+- Exceptions are files which have their own naming conventions (eg Makefile, README)
+
+## Package.json
+
+Plugins are packages, the package definitions of a plugin should include a `consolePlugin`
+map that defines the plugin declarations file, and the plugin integration tests.
+
+``` json
+// File: package.json
+
+{
+  "name": "@console/demo-plugin",
+  "version": "0.0.0-fixed",
+  "description": "Demo plugin for Console web application",
+  "private": true,
+  "dependencies": {
+    "@console/plugin-sdk": "0.0.0-fixed",
+    "@console/shared": "0.0.0-fixed"
+  },
+  "devDependencies": {
+    "@console/internal-integration-tests": "0.0.0-fixed"
+  },
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    "integrationTests": {
+      "demo": [
+        "@console/internal-integration-tests/tests/base.scenario.ts",
+        "integration-tests/tests/demo.scenario.ts"
+      ]
+    }
+  }
+}
+```
+
+## Plugin Decelerations File (`.../src/plugin.tsx`)
+
+The plugin declaration entry file is defined in `package.json` in the `consolePlugin: entry` variable,
```

Comment:
I'd use a dotted path notation, i.e. `consolePlugin.entry` value.

(These aren't variables from JSON file processing point of view.)